server/data/tokenVerificationClient.ts

```diff
@@ -0,0 +1,39 @@
+import type { Request } from 'express'
+import { RestClient, asUser } from '@ministryofjustice/hmpps-rest-client'
+import config from '../config'
+import logger from '../../logger'
+
+export default class TokenVerificationClient extends RestClient {
+  constructor() {
+    super('Token Verification API', config.apis.tokenVerification, logger)
+  }
+
+  verifyToken(request: Request) {
+    const { user, verified } = request
+
+    if (!config.apis.tokenVerification.enabled) {
+      logger.debug('Token verification disabled, returning token is valid')
+      return true
+    }
+
+    if (verified) {
+      return true
+    }
+
+    logger.debug(`Token request for user "${user.username}'`)
+
+    return this.post({ path: `/token/verify` }, asUser(user.token))
+      .then((response: { active: boolean }) => {
+        if (response && response.active) {
+          request.verified = true
+          return true
+        }
+
+        return false
+      })
+      .catch((e) => {
+        logger.debug(`Token verification failed for user "${user.username}'`)
+        return false
+      })
```

Comment:
is there a reason you're rewriting this to not use async await?

Reply:
Personal preference thing - I'm not a really big fan of async/await because of the try/catch block that you end up needing to wrap stuff with. 

This was initially missing that (though I think it was safe at the time because the error was caught in the function itself, and instead nothing was returned). So I just defaulted back to my normal. Happy to switch it to async/await (though I'll do that when moving it to it's own package as discussed on Slack)

Reply:
I think GDS [generally recommended](https://gds-way.digital.cabinet-office.gov.uk/manuals/programming-languages/nodejs/#use-of-async-await) using async await